tests/manage/z_cluster/nodes/test_node_replacement_proactive.py

```diff
@@ -0,0 +1,102 @@
+import logging
+import re
+
+import pytest
+import random
+
+from tests.helpers import get_worker_nodes
+
+from ocs_ci.framework.pytest_customization.marks import tier4a
+from ocs_ci.ocs.resources import pod
+
+from ocs_ci.framework.testlib import (
+    tier4, ManageTest, aws_platform_required, ignore_leftovers, ipi_deployment_required
+)
+from ocs_ci.ocs import (
+    machine, constants, ocp, node
+)
+from tests.sanity_helpers import Sanity
+
+log = logging.getLogger(__name__)
+
+
+@tier4
+@tier4a
+@ignore_leftovers
+@aws_platform_required
+@ipi_deployment_required
+class TestNodeReplacement(ManageTest):
+    """
+    Knip-894 Node replacement - AWS-IPI-Proactive
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+
+        """
+        self.sanity_helpers = Sanity()
+
+    def test_nodereplacement_proactive(self, pvc_factory, pod_factory, dc_pod_factory):
+        """
+        Knip-894 Node Replacement proactive
+
+        """
+
+        # Get worker nodes
+        worker_node_list = get_worker_nodes()
+        log.info(f"Current available worker nodes are {worker_node_list}")
+
+        osd_pods_obj = pod.get_osd_pods()
+        osd_node_name = pod.get_pod_node(random.choice(osd_pods_obj)).name
+        log.info(f"Selected OSD is {osd_node_name}")
+
+        log.info("Creating dc pod backed with rbd pvc and running io in bg")
+        for worker_node in worker_node_list:
+            if worker_node != osd_node_name:
+                rbd_dc_pod = dc_pod_factory(interface=constants.CEPHBLOCKPOOL, node_name=worker_node, size=20)
+                pod.run_io_in_bg(rbd_dc_pod, expect_to_fail=False, fedora_dc=True)
+
+        log.info("Creating dc pod backed with cephfs pvc and running io in bg")
+        for worker_node in worker_node_list:
+            if worker_node != osd_node_name:
+                cephfs_dc_pod = dc_pod_factory(interface=constants.CEPHFILESYSTEM, node_name=worker_node, size=20)
+                pod.run_io_in_bg(cephfs_dc_pod, expect_to_fail=False, fedora_dc=True)
+
+        # Unscheduling node
+        node.unschedule_nodes([osd_node_name])
+        log.info("Getting machine name from specified node name")
+        machine_name = machine.get_machine_from_node_name(osd_node_name)
+        log.info(f"Node {osd_node_name} associated machine is {machine_name}")
+        log.info(f"Deleting machine {machine_name}")
+        machine.delete_machine_and_check_state_of_new_spinned_machine(machine_name)
+        new_machine_list = machine.get_machines()
+        for machines in new_machine_list:
```

Comment:
Is this for getting the new spun node name?
We can get the worker list before machine delete and after machine delete and can find out the new spun machine.

Reply:
this function is used to get new machine name after old is deleted

Reply:
How this functions deals incase of machinset having more than one machines in it?

Reply:
for each worker, master there is separate machine created